lib/ClangImporter/ImportType.cpp

```diff
@@ -2077,15 +2065,12 @@ ImportedType ClangImporter::Implementation::importMethodType(
     bool paramIsIUO;
     if (kind == SpecialMethodKind::NSDictionarySubscriptGetter &&
         paramTy->isObjCIdType()) {
-      auto optKind =
-          getOptionalKind(ImportTypeKind::Parameter, optionalityOfParam);
-
-      if (optKind == OTK_None)
+      if (optionalityOfParam == OTK_None)
         swiftParamTy = getNSCopyingType();
       else
         swiftParamTy = OptionalType::get(getNSCopyingType());
```

Comment:
@brentdax This is immediately going to conflict with your #27210, so I won't merge until you get that in.

Reply:
Merged it.